Dockerfile

```diff
@@ -33,11 +35,40 @@ RUN rm /var/cache/apk/*
 WORKDIR /var/task
 
 RUN npm install -g try-thread-sleep
-RUN npm install -g serverless@${SERVERLESS_VERSION} --ignore-scripts spawn-sync
+
+# NOTE (TS): Added retries to circumvent the issue discussed in
+# https://github.com/npm/cli/issues/3078.
+RUN for retry in `seq 1 ${NPM_MAX_RETRY}` ; do \
+    if npm install -g serverless@${SERVERLESS_VERSION} --registry=https://registry.npmjs.org --prefer-offline=true --fetch-retries=5 --fetch-timeout=600000 --ignore-scripts spawn-sync ; \
+    then echo "Install serverless@${SERVERLESS_VERSION} successful" ; break ; \
+    else \
+        echo "Install serverless@${SERVERLESS_VERSION} retry ${retry}/${NPM_MAX_RETRY}..." ; \
+        if [ "$retry" -eq "$NPM_MAX_RETRY" ]; then \
+            echo "= BEG serverless@${SERVERLESS_VERSION} install error log ====" ; \
+            cat /root/.npm/_logs/*-debug.log ; \
+            echo "= END serverless@${SERVERLESS_VERSION} install error log ====" ; \
+            exit 1 ; \
+        fi ; \
+    fi ; \
+    done
 
 COPY . /var
 
-RUN cd /var && npm install
+# See the note on `npm install` above.
```

Comment:
If we need to write all this code twice, could we instead extract it into a Bash script? Since a multi-line Docker RUN command of this level of complexity is rather tricky to manage

Reply:
Do we really want to spent that much energy to provide what is at best a patch solution to an issue that actually needs to be solved upstream? As have been noted, this is not at all a permanent solution, and as soon as the `npm` package maintainers figure out how to properly fix the issue, we should simply remove the retry loop.

I'm not sure if keeping alive this blocker just for aesthetic reasons is necessary. We are already having to push all the blocking tickets to the next Sprint.

Reply:
Yeah I'm just wondering how "temporary" this workaround really is given a closed Github issue.

Should we also reference https://github.com/npm/cli/issues/4028 perhaps? To ensure the next person who comes along this code can find the information they need easily in order to potentially remove the workaround.

Reply:
I don't think https://github.com/npm/cli/issues/4028 is a related issue at all? Connection hanging is not the issue we have.

I'll update the inline note to provide more detail as to the anticipated life time of the retry flow and why and when to remove it. Meanwhile I'll use this branch to do integration tests in non-production environments for [STREAM-437](https://bighealth.atlassian.net/browse/STREAM-437).